src/gluonts/ext/r_forecast/_univariate_predictor.py

```diff
@@ -136,7 +137,29 @@ def _get_r_forecast(self, data: Dict) -> Dict:
         r_params = self._robjects.vectors.ListVector(self.params)
         vec = self._robjects.FloatVector(data["target"])
         ts = make_ts(vec, frequency=self.period)
-        forecast = self._r_method(ts, r_params)
+
+        if (
+            "feat_dynamic_real" in data
+            and self.method_name == "fourier.arima.xreg"
+        ):
+            import rpy2.robjects.numpy2ri
```

Comment:
@leica2023 out of curiosity: any specific reason for having the import here instead of at the top level?

Reply:
Just tested again by moving line 145 and 147 to the very top, and it worked okay (for some reason, it returned an error in earlier test). Let me move it.

Reply:
Update: just had another two CI tests -- moved the 2 lines up and back down. 

by having these 2 lines on the top, local pytest on the relevant test file succeeded. However, the CI workflow with the updated PR failed and the error happened for the test_r_hierarchical_predictor.py file (irrelevant to this PR, complaining about rpy2 not imported. This reminds of why I moved these two lines down inside the function). After moving them back down, tests ran okay. 

I do not have an answer of why there's a difference having them in different locations. Will try to search and read the documentation to understand how it works behind the stage.